### PR TITLE
fix(ilp): fix spurious errors in Java ILP Client

### PR DESCRIPTION
**Pull request by bluestreak01**
I find that using Os.sleep() produced better performance results in both my custom bench (2x gain) and TSBS. Here is TSBS to help visualise perf delta:

`master`
```
➜  ~ ~/tmp/go/bin/tsbs_load_questdb --file /tmp/data --workers 6
time,per. metric/s,metric total,overall metric/s,per. row/s,row total,overall row/s
1686683872,20965006.79,2.097000E+08,20965006.79,2096500.68,2.097000E+07,2096500.68

Summary:
loaded 345600000 metrics in 16.387sec with 6 workers (mean rate 21090330.55 metrics/sec)
loaded 34560000 rows in 16.387sec with 6 workers (mean rate 2109033.05 rows/sec)
➜  ~ ~/tmp/go/bin/tsbs_load_questdb --file /tmp/data --workers 6
time,per. metric/s,metric total,overall metric/s,per. row/s,row total,overall row/s
1686683894,21845313.82,2.185000E+08,21845313.82,2184531.38,2.185000E+07,2184531.38

Summary:
loaded 345600000 metrics in 15.993sec with 6 workers (mean rate 21609838.92 metrics/sec)
loaded 34560000 rows in 15.993sec with 6 workers (mean rate 2160983.89 rows/sec)
➜  ~ ~/tmp/go/bin/tsbs_load_questdb --file /tmp/data --workers 6
time,per. metric/s,metric total,overall metric/s,per. row/s,row total,overall row/s
1686683916,19205340.45,1.921000E+08,19205340.45,1920534.05,1.921000E+07,1920534.05

Summary:
loaded 345600000 metrics in 18.873sec with 6 workers (mean rate 18311830.32 metrics/sec)
loaded 34560000 rows in 18.873sec with 6 workers (mean rate 1831183.03 rows/sec)
➜  ~ ~/tmp/go/bin/tsbs_load_questdb --file /tmp/data --workers 6
time,per. metric/s,metric total,overall metric/s,per. row/s,row total,overall row/s
1686683941,20336302.46,2.034000E+08,20336302.46,2033630.25,2.034000E+07,2033630.25

Summary:
loaded 345600000 metrics in 17.021sec with 6 workers (mean rate 20304168.40 metrics/sec)
loaded 34560000 rows in 17.021sec with 6 workers (mean rate 2030416.84 rows/sec)
```

`this branch`

```
➜  ~ ~/tmp/go/bin/tsbs_load_questdb --file /tmp/data --workers 6
time,per. metric/s,metric total,overall metric/s,per. row/s,row total,overall row/s
1686684086,23164867.48,2.317000E+08,23164867.48,2316486.75,2.317000E+07,2316486.75

Summary:
loaded 345600000 metrics in 14.764sec with 6 workers (mean rate 23409080.02 metrics/sec)
loaded 34560000 rows in 14.764sec with 6 workers (mean rate 2340908.00 rows/sec)
➜  ~ ~/tmp/go/bin/tsbs_load_questdb --file /tmp/data --workers 6
time,per. metric/s,metric total,overall metric/s,per. row/s,row total,overall row/s
1686684106,23345541.92,2.335000E+08,23345541.92,2334554.19,2.335000E+07,2334554.19

Summary:
loaded 345600000 metrics in 14.729sec with 6 workers (mean rate 23464065.59 metrics/sec)
loaded 34560000 rows in 14.729sec with 6 workers (mean rate 2346406.56 rows/sec)
➜  ~ ~/tmp/go/bin/tsbs_load_questdb --file /tmp/data --workers 6
time,per. metric/s,metric total,overall metric/s,per. row/s,row total,overall row/s
1686684133,20434720.63,2.044000E+08,20434720.63,2043472.06,2.044000E+07,2043472.06

Summary:
loaded 345600000 metrics in 16.440sec with 6 workers (mean rate 21022399.43 metrics/sec)
loaded 34560000 rows in 16.440sec with 6 workers (mean rate 2102239.94 rows/sec)
➜  ~ ~/tmp/go/bin/tsbs_load_questdb --file /tmp/data --workers 6
time,per. metric/s,metric total,overall metric/s,per. row/s,row total,overall row/s
1686684157,26322314.90,2.633000E+08,26322314.90,2632231.49,2.633000E+07,2632231.49

Summary:
loaded 345600000 metrics in 13.059sec with 6 workers (mean rate 26463677.60 metrics/sec)
loaded 34560000 rows in 13.059sec with 6 workers (mean rate 2646367.76 rows/sec)
➜  ~ 
```

### CODE REVIEW

**Comment by questdb-butler** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please update the PR title to match this format:
`type(subType): description`

Where `type` is one of:
`feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, `test`, `ci`, `chore`, `revert`

And: `subType` is one of:
`build`, `sql`, `log`, `mig`, `core`, `ilp`, `pgwire`, `http`, `conf`, `ui`, `wal`

For Example:

```
perf(sql): improve pattern matching performance for SELECT sub-queries
```
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4e48b86eafed362217f0f4a6f4a809038f3f703a
</p>

File: core/src/main/java/io/questdb/std/Os.java [hunk 1]
```diff
@@ -149,13 +149,9 @@ public static void park() {
...
     public static void pause() {
-        if (Os.type != Os.WINDOWS) {
-            LockSupport.parkNanos(1);
-        } else {
-            try {
-                Thread.sleep(0);
-            } catch (InterruptedException ignore) {
-            }
+        try {
+            Thread.sleep(0);
```
Comment (jerrinot) [inline core/src/main/java/io/questdb/std/Os.java:153]: a left-over?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 10 / 11 (90.91%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/Os.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/PlainTcpLineChannel.java|9|9|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 10 / 11 (90.91%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/Os.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/PlainTcpLineChannel.java|9|9|100.00%|

**Review by jerrinot — APPROVED** [review]

**Pull request merged**